notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "    req = sql_client.sql_request(sql)\n",
+    "    req.add_context(\"populateCache\", \"false\")  # run without cacheing results to get a real sense of performance\n",
+    "    req.add_context(\"useCache\", \"false\")  # do not use cached results\n",
+    "    stats = []\n",
+    "    while (iterations>0):\n",
+    "      start = datetime.now()\n",
+    "      sql_client.sql(req)\n",
+    "      end = datetime.now()\n",
+    "      stats.append( (end - start).total_seconds() * 10**3 ) # add run time in milliseconds\n",
+    "      iterations -=1\n",
+    "    return f\"Results = avg:{mean(stats)} ms   min:{min(stats)} ms  max:{max(stats)} ms\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de7c2156-99bf-42bf-858e-537209c20fab",
+   "metadata": {},
+   "source": [
+    "Run the test SQL to see that in the month of November in 2005, American Airlines had the highest percentage of delayed flights on Sundays(7) although the worst average delay occurs on Fridays(5):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42569b72-e018-4576-b8e5-121d2e2a36c5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "SELECT EXTRACT( DOW FROM __time) as day_of_week, \n",
+    "       AVG(DepDelay) FILTER ( WHERE DepDelay>120) as avg_delay,\n",
+    "       MAX(DepDelay) FILTER ( WHERE DepDelay>120) as max_delay,\n",
+    "       ROUND(COUNT(1) FILTER ( WHERE DepDelay>120) * 100.0 / COUNT(1), 1) as percent_delayed\n",
+    "FROM {}\n",
+    "WHERE \"IATA_CODE_Reporting_Airline\" = 'AA'\n",
+    "GROUP BY 1 \n",
+    "ORDER BY 1 ASC\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql.format(\"flights_hour\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d83c7eb1-6d0a-4a08-aa1b-03ec8b755539",
+   "metadata": {},
+   "source": [
+    "### Performance Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dc4c2524-0eba-4bc6-84ed-da3a25aa5fbe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# run 100 queries with each table\n",
+    "use_table=\"flights_hour\"\n",
+    "print (f\"{use_table} {measure_query(sql.format(use_table), 100)}\")\n",
+    "\n",
+    "use_table=\"flights_year\"\n",
+    "print (f\"{use_table} {measure_query(sql.format(use_table), 100)}\")\n",
+    "\n",
+    "use_table=\"flights_year_50k\"\n",
+    "print (f\"{use_table} {measure_query(sql.format(use_table), 100)}\")\n",
+    "\n",
+    "use_table=\"flights_year_IATA\"\n",
+    "print (f\"{use_table} {measure_query(sql.format(use_table), 100)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d0716364-af1f-4eb1-9e75-3ce3c5617344",
+   "metadata": {},
+   "source": [
+    "### Single Concurrency Results\n",
+    "\n",
+    "Given that the characteristics of your laptop or other execution environment are likely different than the `Apple M2 Pro` that this was built on, your results will vary, my results were:\n",
```

Comment:
Suggest you don't need to copy your own results – your bullet points are enough for people to then unpick what happened.

Maybe you could run just one test, then have this explanation, and then run all four afterwards.

Reply:
I'm not sure how consistent the results will be on different laptops. I think it is a good idea to show my results, otherwise, what I say may not make sense with their results.